### PR TITLE
10339: Fix tuntap contants for parisc, powerpc & sparc

### PR DESCRIPTION
**Pull request by mgorny**
## Scope and purpose

This PR fixes `_TUNSETIFF` and `_TUNGETIFF` constant values for parisc, powerpc and sparc architectures. This makes it possible to — at the very least — skip the tests correctly due to lack of permissions, rather than causing the test suite to error on invalid ioctl.

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10339
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [x] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-trac-id

Author: <github_username>, <github_usernames_if_more_authors>
Reviewer: <github_username>, <github_username_if_more_reviewers>
Fixes: ticket:<trac_ticket_number>, ticket:<another_if_more_in_one_PR>

Long description providing a summary of these changes.
(as long as you wish)
```

### CODE REVIEW

**Comment by mgorny** [comment]
I don't think I can satisfy codecov.

File: src/twisted/pair/tuntap.py [hunk 1]
```diff
@@ -36,8 +37,16 @@
...
 _IFNAMSIZ = 16
-_TUNSETIFF = 0x400454CA
-_TUNGETIFF = 0x800454D2
+if (
+    platform.machine() == "parisc"
+    or platform.machine().startswith("ppc")
+    or platform.machine().startswith("sparc")
+):
+    _TUNSETIFF = 0x800454CA  # pragma: nocov
+    _TUNGETIFF = 0x400454D2  # pragma: nocov
```
Comment (adiroiban) [inline src/twisted/pair/tuntap.py]: In twisted we had a zero coverage exception policy via coverage.rc

but I think it should be updated.

Also, as documented on coverage.py site the pragma should be `pragma: no coverage`

https://coverage.readthedocs.io/en/6.3.2/excluding.html

```suggestion
):  # pragma: no coverage
    # We don't have CI for parisc, hence no coverage is expected.
    _TUNSETIFF = 0x800454CA
    _TUNGETIFF = 0x400454D2
```

**Review by adiroiban — APPROVED** [review]
Thanks. I think that we can merge this, with a coverage exception.

File: src/twisted/pair/tuntap.py (hunk 1 above)
Comment (mgorny) [reply src/twisted/pair/tuntap.py]: Thank you! To be honest, I've copied `pragma: nocov` from `src/twisted/web/test/test_http.py`.

Comment (adiroiban) [reply src/twisted/pair/tuntap.py]: Thanks. Those pragma can be updated in another PR if someone is bothered.

**Pull request merged**